_posts/2023-11-20-securing-application-with-other-oidc.adoc

```diff
@@ -0,0 +1,274 @@
+
+== Prerequisite 
+To follow along this guide you will need: 
+
+* Roughly 15-20 minutes
+* Access to an OpenShift cluster (try the https://developers.redhat.com/developer-sandbox[Red Hat Developer Sandbox] for free)
+* https://docs.openshift.com/container-platform/4.13/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+* https://helm.sh/docs/intro/install/[Helm Chart]
+* https://www.okta.com/[Okta OpenID provider]
+* https://github.com/[Github account]
+
+== Example application 
+
+We will use a simple web application in this guide that consists of a single servlet. We will secure this servlet using OIDC. We will use the `simple-webapp-okta` application in this repository. 
+
+In order to deploy this application to OpenShift, we will need to make some changes to the helm chart and add it to a GitHub repository, which OpenShift will pull from. In order to do this, we will need to first fork the https://github.com/wildfly-security-incubator/elytron-examples[elytron-examples] repository. 
+
+Now we can clone the forked elytron-examples repository to your local machine. We will also need to create a new branch, where we will make some changes and commit it to:
+```
+git clone git@github.com:<your_github_username>/elytron-examples.git
+git checkout -b simple-webapp-okta
+cd simple-webapp-okta
+```
+
+== Log Into the OpenShift Cluster
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the OpenShift CLI:
+```
+oc login -u myUserName
+```
+Alternatively, to log in using the API token, navigate to Developer Sandbox, click on your username at the top right corner, copy your login command and paste it on your terminal. The command will be of this format:
+
+```
+oc login --token=myToken --server=myServerUrl
+```
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
+
+== Setting Up the Okta Admin Console 
+In order to secure the application, we will need to access the web console. 
+
+. Visit https://www.okta.com/free-trial/ and under `Workforce Identity Cloud`, sign up or log in to get started. 
+. At the top right corner, click on *Admin* to be redirected to the admin dashboard. 
+. Under the *Add an app to use Single Sign-on* option, click on *Add App* to add the application we want to secure. Click on *Create New App*, choose *OIDC - OpenID Connect* under `Sign-in method` and *Web Application* for `Application type` and click `Next`. 
+. Under *General Settings* add the following configurations: 
+* `App integration name:` *simple-webapp-okta*
+* `Grant type:` Check *Client Credentials* 
```

Comment:
I don't think this needs to be specified, I think we can just leave Authorization Code checked.

Reply:
I had some issues with public client before. I will try again and update it if it works.

Reply:
nvm fixed it! 
This is not the same as public client. 
It should be set to false even if Authorization code is not checked.